src/gameTest/java/mekanism/common/tests/energy/EnergyConversionTest.java

```diff
@@ -0,0 +1,237 @@
+package mekanism.common.tests.energy;
+
+import mekanism.api.Action;
+import mekanism.api.energy.IStrictEnergyHandler;
+import mekanism.common.capabilities.energy.BasicEnergyContainer;
+import mekanism.common.config.MekanismConfig;
+import mekanism.common.integration.energy.forgeenergy.ForgeStrictEnergyHandler;
+import mekanism.common.tests.helpers.EnergyTestHelper;
+import net.minecraft.gametest.framework.GameTest;
+import net.neoforged.neoforge.energy.EnergyStorage;
+import net.neoforged.neoforge.energy.IEnergyStorage;
+import net.neoforged.testframework.annotation.ForEachTest;
+import net.neoforged.testframework.annotation.TestHolder;
+import net.neoforged.testframework.gametest.EmptyTemplate;
+
+@ForEachTest(groups = "energy.conversion")
+public class EnergyConversionTest {
+
+    @GameTest
+    @EmptyTemplate
+    @TestHolder(description = "Tests that energy conversion works properly with various decimal values.")
+    public static void testForgeEnergy(final EnergyTestHelper helper) {
+        double defaultConversion = 2.5;
+        double inverseConversion = 1 / defaultConversion;
+        int capacity = 1_000;
+        int convertedCapacity = (int) (capacity / defaultConversion);
+        helper.startSequence()
+              //Ensure the config starts out at the default value
+              .thenExecute(() -> MekanismConfig.general.forgeConversionRate.set(defaultConversion))
+
+              // WRAPPING STRICT ENERGY TO FORGE ENERGY
+
+              //Test against a small empty container
+              .thenMap(() -> helper.createForgeWrappedStrictEnergyHandler(0, capacity))
+              .thenExecute(handler -> {
+                  int extracted = handler.extractEnergy(capacity, false);
+                  helper.assertValueEqual(extracted, 0, "extracted energy");
+                  int accepted = handler.receiveEnergy(capacity, false);
+                  helper.assertValueEqual(accepted, convertedCapacity, "accepted energy");
+                  helper.assertValueEqual(handler.getEnergyStored(), convertedCapacity, "stored energy");
```

Comment:
these checks should not use wrappers.
interact via the wrappers, check with the unwrapped storage. Applies to all these tests